env0/data_project.go

```diff
@@ -81,7 +86,31 @@ func dataProjectRead(ctx context.Context, d *schema.ResourceData, meta interface
 	return nil
 }
 
-func getProjectByName(name interface{}, meta interface{}) (client.Project, error) {
+func filterByParentProjectName(name string, parentName string, projects []client.Project, meta interface{}) ([]client.Project, error) {
+	if len(parentName) > 0 {
```

Comment:
It'd be cleaner to take this condition out of this function, as this function should be called only if using the parent name attribute

Reply:
@yaronya - cleaned it up. It should make more sense now.